aten/src/ATen/native/native_functions.yaml

```diff
@@ -2514,11 +2522,25 @@
     CPU: _sinh_out_cpu
     CUDA: _sinh_out_cuda
 
+# Returns a copy of this `Variable` that is detached from its autograd graph
+# and has a blank version. This method is OK to call if the `Variable` is a
+# view.
+# NOTE: Previously, if we change the tensor metadata (e.g. sizes / strides /
+# storage / storage_offset) of a tensor created from `detach()`, those metadata
+# in the original tensor will also be updated. However, the new behavior is that
+# those metadata changes to the detached tensor will not update the original tensor
+# anymore, and in the `detach()` function we need to set `allow_tensor_metadata_change_`
+# to false to make such changes explicitly illegal, in order to prevent users from
+# changing metadata of the detached tensor and expecting the original tensor to also
+# be updated.
 - func: detach(Tensor self) -> Tensor
   use_c10_dispatcher: full
   supports_named_tensor: True
   variants: function, method
 
+# Like `detach()`, but removes this `Variable` in-place. This method may
```

Comment:
Could you update this doc to mention that this should be forbidden on Variable that are the base of other Variables and a TODO: fix.
This is not a hard bug, but is very misleading and does not really make sense to allow.
See below for more details.

This is a problem if we then change this other view inplace, then the rebase_history is going to tinker with the newly detached Variable. In practice, linking gradient to the detached base which now requires_grad=True again but does not link to its original history anymore.
A clean way to fix this would be to forbid (or at the very least warn very loudly) "rebase_history" on a base that does not require gradients.

Reply:
I can do that in a separate PR, but I don't think this bug is related to this PR, is it?

Reply:
No definitely not introduced here.
Just that I stumbled upon it last week and didn't had time to open an issue about it, so I wanted to write this down somewhere if possible.